src/components/BulkEditModal/BulkEditModal.test.tsx

```diff
@@ -0,0 +1,220 @@
+//     enabled: true,
+//     labels: [],
+//     probes: [32, 42],
+//     settings: {
+//       ping: {
+//         ipVersion: IpVersion.V4,
+//         dontFragment: false,
+//       },
+//     },
+//     basicMetricsOnly: false,
+//   },
+//   {
+//     job: '',
+//     alertSensitivity: 'none',
+//     target: '',
+//     frequency: 60000,
+//     timeout: 3000,
+//     enabled: true,
+//     labels: [],
+//     probes: [32, 42],
+//     settings: {
+//       ping: {
+//         ipVersion: IpVersion.V4,
+//         dontFragment: false,
+//       },
+//     },
+//     basicMetricsOnly: false,
+//   },
+// ]);
+
+const renderBulkEditModal = (action: 'add' | 'remove' | null, selectedChecks: () => FilteredCheck[]) => {
+  const instance = {
+    api: getInstanceMock(),
+    metrics: {},
+    logs: {},
+  } as GrafanaInstances;
+  const meta = {} as AppPluginMeta<GlobalSettings>;
+
+  render(
+    <InstanceContext.Provider value={{ instance, loading: false, meta }}>
+      <SuccessRateContextProvider checks={[]}>
+        <BulkEditModal
+          onDismiss={onDismiss}
+          onSuccess={onSuccess}
+          onError={onError}
+          selectedChecks={selectedChecks}
+          instance={instance}
+          action={action}
+          isOpen={true}
+        />
+      </SuccessRateContextProvider>
+    </InstanceContext.Provider>
+  );
+
+  return instance;
+};
+
+test('shows the modal', async () => {
+  renderBulkEditModal('add', selectedChecksSingleProbe);
+  const title = await screen.findByText('Add probes to 2 selected checks');
+  const probes = await screen.findAllByTestId('probe-button');
+  expect(title).toBeInTheDocument();
+  expect(probes).toHaveLength(2);
+});
+
+test('successfully adds probes', async () => {
+  const instance = renderBulkEditModal('add', selectedChecksSingleProbe);
+  await act(async () => {
+    const burritoProbe = await screen.findByText('burritos');
+    const tacoProbe = await screen.findByText('tacos');
+    userEvent.click(burritoProbe);
+    userEvent.click(tacoProbe);
+    const submitButton = await screen.findByText('Submit');
+    userEvent.click(submitButton);
+  });
+  expect(instance.api?.bulkUpdateChecks).toHaveBeenCalledWith([
+    {
+      job: '',
+      alertSensitivity: 'none',
+      target: '',
+      frequency: 60000,
+      timeout: 3000,
+      enabled: true,
+      labels: [],
+      probes: [32, 42],
+      settings: {
+        ping: {
+          ipVersion: IpVersion.V4,
+          dontFragment: false,
+        },
+      },
+      basicMetricsOnly: false,
+    },
+    {
+      job: '',
+      alertSensitivity: 'none',
+      target: '',
+      frequency: 60000,
+      timeout: 3000,
+      enabled: true,
+      labels: [],
+      probes: [42, 32],
+      settings: {
+        ping: {
+          ipVersion: IpVersion.V4,
+          dontFragment: false,
+        },
+      },
+      basicMetricsOnly: false,
+    },
+  ]);
+});
+
+// test.only('successfully removes probes', async () => {
```

Comment:
leaving as todo?

Reply:
Actually could use your help troubleshooting this test, there's some RTL thing I'm missing